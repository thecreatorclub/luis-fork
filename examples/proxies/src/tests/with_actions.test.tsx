import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { observable } from 'mobx';
import { observer } from 'mobx-react';

class State {
  @observable number = 1;
}

type Props = {
  state: State;
};

@observer
class Component extends React.Component<Props, State> {
  state = { number: 1 };

  render() {
    return (
      <div>
        <div>Current mobx number: {this.props.state.number}</div>
        <div>Current state number: {this.state.number}</div>
        <button onClick={() => this.setState({ number: this.state.number + 1 })}>Increase</button>
      </div>
    );
  }
}

describe('With Actions', () => {
  function componentWithData(state: State = new State()) {
    return <Component state={state} />;
  }

  it('renders correctly', () => {
    const state = new State();
    state.number = 9;

    const wrapper = renderer.create(componentWithData(state));
    expect(wrapper).toMatchSnapshot('changed to 9');
    //state.number = 10;
    wrapper.root.findByType('button').props.onClick({});
    // wrapper.find('button').simulate('click');
    state.number = 10;
    expect(wrapper).toMatchSnapshot('changed to 10');
  });
});
