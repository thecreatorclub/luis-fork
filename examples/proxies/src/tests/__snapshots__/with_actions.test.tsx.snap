// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Actions renders correctly: changed to 9 1`] = `
<div>
  <div>
    Current mobx number: 
    9
  </div>
  <div>
    Current state number: 
    1
  </div>
  <button
    onClick={[Function]}
  >
    Increase
  </button>
</div>
`;

exports[`With Actions renders correctly: changed to 10 1`] = `
<div>
  <div>
    Current mobx number: 
    10
  </div>
  <div>
    Current state number: 
    2
  </div>
  <button
    onClick={[Function]}
  >
    Increase
  </button>
</div>
`;
